content/api/cypress-api/custom-queries.md

```diff
@@ -0,0 +1,385 @@
+---
+title: Custom Queries
+---
+
+Starting in Cypress 12, Cypress comes with its own API for creating custom
+queries. The built in Cypress queries use the very same API that's explained
```

Comment:
Updated this top section to make it clearer. Also added more hints pointing people towards custom commands instead of queries, since that's what 95% of people will want.

> Queries are a type of command, used for *querying* the state of your
application. They are different from other commands in that they follow three
important rules:
> 1. Queries are _synchronous._ They do not return or await promises.
> 2. Queries are _retriable._ Once you return the inner function, Cypress takes
   control, handling retries on your behalf.
> 3. Queries are _idempotent._ Once you return the inner function, Cypress will
   invoke it repeatedly. Invoking the inner function multiple times must not
   change the state of your application.

> ### \<Info box>
> If you want to chain together existing Cypress commands as a shortcut, you
probably want to write a [custom command](/api/cypress-api/custom-command)
instead.

> You'll also want to write a command instead of a query if your method needs to
be asynchronous, or it shouldn't be called more than once.